src/pages/[platform]/build-a-backend/server-side-rendering/nextjs-v5-to-v6-migration-guide/index.mdx

```diff
@@ -0,0 +1,361 @@
+
+## Using the API category in v6
+
+Amplify v6 offers [two specialized GraphQL API clients](/[platform]/build-a-backend/graphqlapi/connect-from-server-runtime/#step-1---choose-the-correct-graphql-api-client-for-nextjs-server-runtimes) for Next.js server runtimes (imported from `@aws-amplify/adapter-nextjs/api`) that you should use depending whether you retrieve the user tokens using [`cookies`](https://nextjs.org/docs/app/api-reference/functions/cookies) or [`NextRequest` and `NextResponse`](https://nextjs.org/docs/app/api-reference/functions/next-request). We will demonstrate the migration using the request/response-based server client to reflect what was previously available in v5.
+
+First add server-side client generation to [your `amplifyServerUtils` file:](/[platform]/build-a-backend/server-side-rendering/nextjs-v5-to-v6-migration-guide/#2-server-side-configuration)
+
+```ts
+import { createServerRunner } from '@aws-amplify/adapter-nextjs';
+import config from '@/amplifyconfiguration.json';
+
+export const { runWithAmplifyServerContext } = createServerRunner({
+  config
+});
+
+export const reqResBasedClient = generateServerClientUsingReqRes({
+  config
+});
+```
+
+Now update the implementation in your server-side-rendered component:
+
+<Columns columns="2">
+  <div>
+    **V5**
+
+    ```ts
+    import { withSSRContext } from 'aws-amplify';
+    import { listTodos } from './graphql/queries';
+
+    const getServerSideProps = async ({ req }) => {
+      const SSR = withSSRContext({ req });
+      const { data } = await SSR.API.graphql({ query: listTodos });
+
+      return {
+        props: {
+          posts: data.listTodos.items
+        }
+      };
+    };
+    ```
+
+  </div>
+  <div>
+    **V6**
+
+    ```ts
+    import { 
+      reqResBasedClient,
+      runWithAmplifyServerContext
+    } from '@/utils/amplifyServerUtils';
+    import { listTodos } from './graphql/queries';
+
+    const getServerSideProps = async ({ req, res }) => {
+      const data = await runWithAmplifyServerContext({
+        nextServerContext: { request: req, response: res },
+        operation: async (contextSpec) => {
+          return reqResBasedClient.graphql(contextSpec, {
+            query: listTodos
+          });
+        }
+      });
+
+      return {
+        props: {
+          todos: data.listTodos.items
+        }
+      };
+    };
+    ```
+
+  </div>
+</Columns>
+
+## Migrating from DataStore to API in a server context
+
+DataStore is no longer supported within a server context in v6. You will need to use the GraphQL API client instead. 
+
+For a successful transition to using the API category in the server context, you will need to be aware of [how conflict resolution works](https://docs.aws.amazon.com/appsync/latest/devguide/conflict-detection-and-sync.html#conflict-detection-and-resolution), paying particular attention to `auto-merge`. This will require the following adjustments to standard GraphQL operations:
```

Comment:
Instead of "paying particular attention to auto-merge", maybe just note that for datastore API 's, auto-merge is the default and are what is assumed in the examples below.